### PR TITLE
drivers: regulator: pca9420: fix support for DVS modes

### PR DESCRIPTION
**Pull request by danieldegrasse**
Fix support for DVS modes, as the dvs_state_set implementation previously had the mask and value parameters in the i2c_reg_update_byte_dt function swapped.

Also, record the active DVS state and update the voltage get/set and regulator enable/disable function to target the active DVS mode. This will enable a user to configure multiple run modes, and modify target voltages for those modes when in the new mode. When a user is utilizing the MODESEL pins, update the active state but return an error so that the application can still edit settings for the new DVS mode once it has reconfigured the appropriate pins.

### CODE REVIEW

File: drivers/regulator/regulator_pca9420.c [hunk 1]
```diff
@@ -439,7 +467,12 @@ static int regulator_pca9420_common_init(const struct device *dev)
 			DT_INST_PROP(inst, nxp_enable_modesel_pins),           \
 	};                                                                     \
                                                                                \
-	DEVICE_DT_INST_DEFINE(inst, regulator_pca9420_common_init, NULL, NULL, \
+	static struct regulator_pca9420_common_data data_##inst = {            \
+		.dvs_state = 0U,                                               \
+	};                                                                     \
```
Comment (gmarull) [inline drivers/regulator/regulator_pca9420.c:472]: zero init for statics is redundant

Comment (danieldegrasse) [reply drivers/regulator/regulator_pca9420.c]: good point. I've removed this.

**Review by gmarull — APPROVED** [review]

**Review by aasinclair — APPROVED** [review]

**Comment by danieldegrasse** [comment]
Rebased to pick up fix for CI failure.

**Comment by gmarull** [comment]
> Rebased to pick up fix for CI failure.

interesting you didn't loose approvals :-)

**Comment by danieldegrasse** [comment]
> interesting you didn't loose approvals :-)

I noticed that as well- perhaps github is getting smarter about when to dismiss them

**Pull request merged**